### PR TITLE
limel-picker: add default searcher function for simple client-side filtering

### PR DESCRIPTION
**Pull request by adrianschmidt**
## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/picker/examples/picker-empty-suggestions.tsx [hunk 1]
```diff
@@ -56,8 +59,11 @@ export class PickerExample {
             if (query === '') {
                 // Simulate some network delay
                 setTimeout(() => {
-                    resolve([]);
+                    const result = this.allItems.slice(8, 10);
+                    resolve(result);
                 }, NETWORK_DELAY);
+
+                return;
```
Comment (jgroth) [inline src/components/picker/examples/picker-empty-suggestions.tsx:66]: We could probably improve this as well if we want to get rid of the `new Promise()` syntax, similar to the other comment I had. The simulated network delay can be done something like this
```ts
await new Promise(resolve => setTimeout(resolve, NETWORK_DELAY));
```

File: src/components/picker/picker.tsx [hunk 2]
```diff
@@ -556,13 +570,44 @@ export class Picker {
...
+
+                return searchText.includes(query.toLowerCase());
+            });
+
+            return resolve(
+                filteredItems.slice(0, DEFAULT_SEARCHER_MAX_RESULTS),
+            );
+        });
+    };
+
```
[the remark this answers is not in the document]
Comment (adrianschmidt) [reply src/components/picker/picker.tsx:610]: Yeah, I took it completely from the basic example.

Maybe they had to be more complicated when we used the mocked network delay, and could have been simplified when we removed that?

Thanks for this improvement!

File: src/components/picker/examples/picker-empty-suggestions.tsx (hunk 1 above)
Comment (adrianschmidt) [reply src/components/picker/examples/picker-empty-suggestions.tsx]: ⚡ 4d4e4e46b5dd2f45eb94a7f6516a5cf21c47a363

**Review by jgroth — APPROVED** [review]

**Pull request merged**